Doc/library/sqlite3.rst

```diff
@@ -2440,15 +2436,16 @@ Here's an example of a factory that return a :class:`~collections.namedtuple`.
    >>> row.b   # Attribute access.
    2
 
-As an optimisation, decorate :func:`!_make_cls` with
-:func:`!functools.lru_cache` to avoid creating multiple named tuples
-with identical column spec.
-See :func:`functools.lru_cache` for more details.
-
 With some adjustments, the above recipe can be adapted to use a
 :class:`~dataclasses.dataclass`, or any other custom class,
 instead of a :class:`~collections.namedtuple`.
 
+As an exercise left for the reader, the above example can be optimised
+by extracting to a new function the code that create the ``fields`` variable.
+Then decorate that new function with :func:`functools.lru_cache` to avoid
+creating multiple classes ``cls`` from identical column specs.
```

Comment:
```suggestion
by extracting the code that create the ``fields`` variable to a new function,
and decorating it with :func:`functools.lru_cache`.
This avoids creating multiple class:`~collections.namedtuple` classes
from identical column specs.
```

* Split the sentence at a more suitable point to avoid implying, as it does now, that pulling the code out to a function is the optimization, as opposed to using `functools.lru_cache`, and also cleanly separate the "how" from the "why" it works, and is also more concise overall.
* In the first line, flip the order to be more idiomatic and read more naturally in English
* In the second line, fix the verb form agreement (which previously mixed imperative with present participle).
* In the third line, fix the grammar/word order and refer specifically to the class factory (since we suggested replacing the reference above with the glossary term).

Reply:
Obsolete, as this is now removed.